notion-distcopy/src/main/scala/com/ambiata/notion/distcopy/DistCopyJob.scala

```diff
@@ -162,7 +163,11 @@ class DistCopyMapper extends Mapper[NullWritable, Mapping, NullWritable, NullWri
           )
         }.retry(retryCount, retryHandler)
         _               = println(s"Moving: $tmpDestination ===> $destination")
-        _               <- S3Action.fromRIO((destination.dirname.mkdirs >> HdfsPath.fromPath(tmpDestination).move(destination)).run(context.getConfiguration))
+        _               <- S3Action.fromRIO((for {
+          _ <- destination.dirname.mkdirs
+          f <- HdfsPath.fromPath(tmpDestination).determineFile
```

Comment:
This seems sub-optimal, any reason you can't overwrite without overhead/race of determine?